controllers/operatorpolicy_controller.go

```diff
@@ -205,6 +205,15 @@ func (r *OperatorPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	policy := &policyv1beta1.OperatorPolicy{}
 	watcher := opPolIdentifier(req.Namespace, req.Name)
 
+	opLog.Info("Reconciling OperatorPolicy")
+
+	defer func() {
+		// ACM-14617: This log is to help identify if a deferred function call is getting stuck on
+		// a lock. This message must be at INFO level because the issue may not be reproduced if the
+		// log level is adjusted *after* the problem occurs.
+		opLog.Info("All deferred functions for the reconcile are complete")
```

Comment:
Instead of this, could it be the reconcile complete log message? If you define the variables outside of this function scope just above, the defer function should pick up the latest values of `finalErr` and etc.

Reply:
I originally tried something like that, but I realized it would be useful to have a "bottom of reconcile" kind of log message, in addition to this "everything actually complete" message - this would help us spot if the deferred cleanup steps are what is hanging. So it was unnecessary to pre-define any variables.